CONTRIBUTING.md

```diff
@@ -101,14 +101,46 @@ This starts the local preview on [localhost:8080](http://localhost:8080). You ca
+We encourage samples for all the topics in .NET documentation. Readers can download and run the samples. They
+can explore the code and experiment with their own changes.
 
-## Contributor License Agreement
+All samples should be complete applications or libraries. Where the sample creates a library,
+it should include unit tests or an application that lets readers run the code.
 
-You must sign the [.NET Foundation Contribution License Agreement (CLA)](http://cla2.dotnetfoundation.org) before your PR is merged. This is a one-time requirement for projects in the .NET Foundation. You can read more about [Contribution License Agreements (CLA)](http://en.wikipedia.org/wiki/Contributor_License_Agreement) on Wikipedia.
+The samples are organized in directories that match the directories under
+docs in this repo:
 
-The agreement: [net-foundation-contribution-license-agreement.pdf](https://cla2.dotnetfoundation.org/cladoc/net-foundation-contribution-license-agreement.pdf)
+- `getting-started` contains getting started samples. 
 
-You don't have to do this up-front. You can simply clone, fork, and submit your PR as usual. When your PR is created, it is classified by a CLA bot. If the change is trivial (for example, you just fixed a typo), then the PR is labeled with `cla-not-required`. Otherwise, it's classified as `cla-required`. Once you signed the CLA, the current and all future pull requests will be labeled as `cla-signed`.
\ No newline at end of file
+   * Each top-level section of our documentation contains a "getting-started" or "tutorials" section.  These topics and their associated samples represent the first look(s) at a topic area. If your sample's goal is to help beginners get started with the platform or the technology, place your sample under `samples/getting-started`. There should be a directory that matches the section for your tutorial (such as `samples/getting-started/csharp`), but if not, simply create a new folder.
+
+- `core-projects` contains .NET Core samples. These are referenced by topics under `/docs/core`.
+
+   * The `core-projects` directory contains samples that highlight .NET Core. The purpose of your sample should be to teach developers something about .NET Core. This includes the framework packaging, the new tooling, or the cross-platform experience.
+
+- `csharp-language` contains C# language samples. These are referenced by topics under `docs/csharp`.
+
+   * The `csharp-language` directory contains samples where the purpose is to explain the C# language. While these samples will use frameworks and libraries, their focus is on the C# language.
+
+ - `linq` contains all the 101 LINQ samples that were originally created when LINQ was developed. These don't have topics associated with them yet.
+
+   * The `linq` directory contains samples that are specific to understanding the concepts and implementation behind Language Integrated Query (LINQ). 
```

Comment:
yes, each has a quick summary, and more details about that directory. (Admittedly there aren't many more details.)